### PR TITLE
Migrate: skip the automigration for gf markdown when user isn't using mdx

### LINKED ISSUE #22167: [Bug]: Found a 'github-flavored-markdown-mdx' migration even without mdx files in the project
### Describe the bug

There seems to be a bug where updating or running:

`npx storybook@next automigrate`

Always offers the migration to github-flavored-markdown-mdx even when there are no mdx or jsx files in the project; I'm talking about an EMPTY project. Storybook 7 is supposed to use MDX2 and no longer supports GFM and that you recommend installing it as a development dependency if you use its features. I wonder what features? How can I use MDX2 without GFM which should I avoid using? If it is assumed that the package isn't provided by default during migration, then the storybook can be used without it.

### To Reproduce

New vite project

`npm create vite@latest` 

Select react project.

`npx storybook@latest init`

This command does not trigger  'github-flavored-markdown-mdx' migration

Then:

`npx storybook@next automigrate` or `npx storybook@latest upgrade`

Found a 'github-flavored-markdown-mdx' migration, even if you delete all files in the project. How is this detected?

### System

```shell
Environment Info:

  System:
    OS: Linux 5.15 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
  Binaries:
    Node: 19.8.1 - ~/.nvm/versions/node/v19.8.1/bin/node
    npm: 9.5.1 - ~/.nvm/versions/node/v19.8.1/bin/npm
  Browsers:
    Chrome: 110.0.5481.177
  npmPackages:
    @storybook/addon-essentials: ^7.0.6 => 7.0.6 
    @storybook/addon-interactions: ^7.0.6 => 7.0.6 
    @storybook/addon-links: ^7.0.6 => 7.0.6 
    @storybook/blocks: ^7.0.6 => 7.0.6 
    @storybook/react: ^7.0.6 => 7.0.6 
    @storybook/react-vite: ^7.0.6 => 7.0.6 
    @storybook/testing-library: ^0.0.14-next.2 => 0.0.14-next.2
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ndelangen**
Closes https://github.com/storybookjs/storybook/issues/22167

## What I did

I added a check if the user is actually using MDX files.
If they do not.. we should skip the automigration.

## How to test

Run the automigration on a project that has no MDX files

### CODE REVIEW

File: code/lib/cli/src/automigrate/fixes/mdx-gfm.ts [hunk 1]
```diff
@@ -22,9 +25,27 @@ export const mdxgfm: Fix<Options> = {
       return null;
     }
 
+    const hasMDXFiles = await mainConfig?.stories?.reduce(async (acc, item) => {
```
Comment (yannbf) [inline code/lib/cli/src/automigrate/fixes/mdx-gfm.ts:28]: I wonder if you could just do the same as this automigration: https://github.com/storybookjs/storybook/blob/next/code/lib/cli/src/automigrate/fixes/mdx-1-to-2.ts#L49

Comment (ndelangen) [reply code/lib/cli/src/automigrate/fixes/mdx-gfm.ts:28]: That seems worse? considering it's not using the glob patterns set in `main.ts`, and will ignore any MDX files not following the `.stories.mdx` pattern.

**Comment by ndelangen** [comment]
@shilman @yannbf can you review / approve?

File: code/lib/cli/src/automigrate/fixes/mdx-gfm.ts (hunk 1 above)
Comment (yannbf) [reply code/lib/cli/src/automigrate/fixes/mdx-gfm.ts:28]: Alright sounds good, maybe the other automigration should follow this pattern then

**Review by yannbf — APPROVED** [review]
LGTM!!

**Pull request merged**